scripts/upgrade.sh

```diff
@@ -35,7 +35,8 @@ EXCLUDE="@types/express @types/react @types/react-dom eslint node-fetch react re
 # --reject - Exclude packages matching the given string
 # --target - Determines the version to upgrade to
 # "minor" - Upgrade to the highest minor version without bumping the major version
-npx npm-check-updates --workspaces --root --upgrade --reject "$EXCLUDE" --target minor
+# `enginesNode` makes sure that packages can be run against the node requirement specified in the monorepo "engines.node"
+npx npm-check-updates --workspaces --root --upgrade --reject "$EXCLUDE" --target minor --enginesNode
```

Comment:
Added `--enginesNode` option to `npm-check-updates`, which makes sure that package versions match the requirement in `engines.node` in `package.json`

This was required so that we didn't increment to `npm` version 11, which does not work on lower versions of Node20 or Node18 at all (which we still support until v4)

We may want to wait a few weeks to let the dust settle on npm 11, but either way we should probably always have this `--enginesNode` check to prevent upgrades that would not work for our desired Node version